### ISSUE TITLE
[Feature] Add Model Hooks for Accessing and Customizing Model Activations

### ISSUE THREAD

**Issue by shuyhere** [sgl-project/sglang#3266]
### Checklist

- [x] 1. If the issue you raised is not a feature but a question, please raise a discussion at https://github.com/sgl-project/sglang/discussions/new/choose Otherwise, it will be closed.
- [x] 2. Please use English, otherwise it will be closed.

## Description
It would be beneficial to introduce model hooks that allow users to access and modify model activations. This feature would enable greater flexibility for tasks such as visualization, debugging, and custom processing of intermediate representations.

## Use case
* Extract intermediate outputs for interpretability analysis, such as [LogitLens-style investigations](https://www.lesswrong.com/posts/AcKRB8wDpdaN6v6ru/interpreting-gpt-the-logit-lens).
* Expose internal activations, enabling users to cache activations and implement functions to edit, remove, or replace them dynamically during inference, for example [representation engineering](https://github.com/andyzoujm/representation-engineering).

While this may introduce some performance overhead, it would enhance interpretability research and enable efficient model editing.

## model hook resources
* [Pytorch hook](https://pytorch.org/docs/stable/generated/torch.Tensor.register_hook.html)
* [Transformerlens](https://github.com/TransformerLensOrg/TransformerLens)
## related issues and use case
* https://github.com/vllm-project/vllm/issues/4084
* https://github.com/vllm-project/vllm/issues/11397
* https://github.com/vllm-project/vllm/issues/8278

**Comment by jalaln06** [comment]
Has this feature been added?

**Comment by CurryxIaoHu** [comment]
Do we have this function now? Does anybody know how to do it?

**Comment by YooSungHyun** [comment]
I'm very interested in KL divergence, so I need the logits used for training LLMs. Is there any way to access them?

**Comment by MingningShao** [comment]
Why issue closed ? Dose this feature implemented? Seems no ???
